assets/src/data/model/ticket/status-helper.js

```diff
@@ -25,31 +25,43 @@ const assertTicketEntity = ( TicketEntity ) => {
 /**
  * @function
  * @param {Object} TicketEntity model object
+ * @param {boolean} includeArchived if true will not filter out archived entities
  * @return {boolean} true if ticket is currently available for purchase
  */
-export const isOnSale = ( TicketEntity ) => {
-	return ! isArchived( TicketEntity ) &&
-		TicketEntity.startDate.diffNow() < 0 &&
-		TicketEntity.endDate.diffNow() > 0;
+export const isOnSale = ( TicketEntity, includeArchived = false ) => {
+	return (
+		( includeArchived && assertTicketEntity( TicketEntity ) ) ||
+		( ! includeArchived && ! isArchived( TicketEntity ) )
+	) &&
+	TicketEntity.startDate.diffNow() < 0 &&
+	TicketEntity.endDate.diffNow() > 0;
 };
 
 /**
  * @function
  * @param {Object} TicketEntity model object
+ * @param {boolean} includeArchived if true will not filter out archived entities
  * @return {boolean} true if ticket can no longer be purchased
  */
-export const isExpired = ( TicketEntity ) => {
-	return ! isArchived( TicketEntity ) &&
-		TicketEntity.endDate.diffNow() < 0;
+export const isExpired = ( TicketEntity, includeArchived = false ) => {
+	return (
+		( includeArchived && assertTicketEntity( TicketEntity ) ) ||
+		( ! includeArchived && ! isArchived( TicketEntity ) )
+	) &&
+	TicketEntity.endDate.diffNow() < 0;
 };
 
 /**
  * @function
  * @param {Object} TicketEntity model object
+ * @param {boolean} includeArchived if true will not filter out archived entities
  * @return {boolean} true if tickets sold meets or exceeds available quantity
  */
-export const isSoldOut = ( TicketEntity ) => {
-	if ( isArchived( TicketEntity ) ) {
+export const isSoldOut = ( TicketEntity, includeArchived = false ) => {
+	if (
+		( includeArchived && ! assertTicketEntity( TicketEntity ) ) ||
+		( ! includeArchived && isArchived( TicketEntity ) )
+	) {
```

Comment:
I have the same feedback here as the status-helper for date-times regarding extracting out this common condition into it's own function that you can call here and elsewhere (to keep code DRY).